### PR TITLE
generator can make packagejson

### PR DESCRIPTION
**Pull request by ericanderson**

### CODE REVIEW

File: packages/cli/src/commands/typescript/generate/handleGenerate.mts [hunk 1]
```diff
@@ -124,34 +128,68 @@ async function generateClientSdk(
...
+    }
+
+    if (!args.asPackage) {
+      await generateSourceFiles(args, ontology, createNormalFs());
+      return true;
+    }
+
+    const packageName = args.packageName;
+    if (!packageName) throw new Error("Package name is require");
+    const version = args.version;
```
Comment (ssanjay1) [inline packages/cli/src/commands/typescript/generate/handleGenerate.mts:145]: Is version not required? If it is, should we do an error check here as well?

File: packages/cli/src/commands/typescript/generate/handleGenerate.mts [hunk 2]
```diff
@@ -161,7 +199,143 @@ async function generateClientSdk(
...
+    strict: true,
+
+    skipLibCheck: true,
+  };
+
+  const compilerOptions = packageType === "commonjs"
+    ? {
+      ...commonTsconfig,
+      module: "commonjs",
+      // moduleResolution: "Node",
```
Comment (ssanjay1) [inline packages/cli/src/commands/typescript/generate/handleGenerate.mts:223]: nit: remove commented out lines

File: packages/cli/src/commands/typescript/generate/handleGenerate.mts [hunk 3]
```diff
@@ -161,7 +199,143 @@ async function generateClientSdk(
...
+      target: "es2018",
+      // declaration: false,
+    }
+    : {
+      ...commonTsconfig,
+      module: "NodeNext",
+      // moduleResolution: "nodenext",
```
Comment (ssanjay1) [inline packages/cli/src/commands/typescript/generate/handleGenerate.mts:230]: nit: commented lines

File: packages/cli/src/commands/typescript/generate/handleGenerate.mts [hunk 4]
```diff
@@ -161,7 +199,143 @@ async function generateClientSdk(
...
+          sdkVersion === 2 ? "" : "/index"
+        }.js`,
+      },
+    },
+    scripts: {
+      prepack:
+        `tsc -p ${esmPrefix}/tsconfig.json && tsc -p ${commonjsPrefix}/tsconfig.json`,
+      check: "npm exec attw $(npm pack)",
+    },
+    // type: type === "root" ? undefined : type,
```
Comment (ssanjay1) [inline packages/cli/src/commands/typescript/generate/handleGenerate.mts:277]: Do we need this?

**Review by ssanjay1 — APPROVED** [review]
Approved, left couple of nit comments

File: packages/cli/src/commands/typescript/generate/handleGenerate.mts (hunk 1 above)
Comment (ericanderson) [reply packages/cli/src/commands/typescript/generate/handleGenerate.mts:145]: version is required already in the yargs

**Pull request merged**